### PR TITLE
Updated the conda install commands to specify the channel

### LINKED ISSUE #1718: Update conda install commands to specify channel
- All conda install commands should specify the channel to use (`conda install -c conda-forge featuretools`)
  - https://github.com/alteryx/featuretools/search?q=conda+install 
- For example, [our botocore error message](https://github.com/alteryx/featuretools/blob/9814869c7ec50d0313dd590b079c37c0fd24765a/featuretools/utils/s3_utils.py#L59), does not specify the channel

### PR DESCRIPTION
**Pull request by kushal-gopal**
Fixes #1718

Updated utils/s3_utils.py file by specifying the channel for the three error messages - BOTO3_ERR_MSG, BOTOCORE_ERR_MSG, SMART_OPEN_ERR_MSG.

One thing which requires review is that I found package name for smart-open in conda-forge to be smart_open.
https://anaconda.org/conda-forge/smart_open

### CODE REVIEW

**Comment by kushal-gopal** [comment]
@gsheni,

Thanks for reviewing.
I did update the docs/source/release_notes.rst file and please let me know if I need to do anything more.
Thanks again for this opportunity as this is my first open source contribution. So yayy!

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -7,13 +7,14 @@ Future Release
...
     * Fixes
+        * Updated the conda install commands to specify the channel (:pr:`1718`)
     * Changes
     * Documentation Changes
         * Add time series guide (:pr:`1896`)
     * Testing Changes
 
     Thanks to the following people for contributing to this release:
-    :user:`tamargrey`
+    :user:`tamargrey`, :user:`kushal-gopal`,
```
Comment (gsheni) [inline docs/source/release_notes.rst:17]:
```suggestion
    :user:`tamargrey`, :user:`kushal-gopal`
```

**Review by gsheni — DISMISSED** [review]
This looks good!

File: docs/source/release_notes.rst [hunk 2]
```diff
@@ -7,13 +7,14 @@ Future Release
 ==============
     * Enhancements
     * Fixes
+        * Updated the conda install commands to specify the channel (:pr:`1718`)
```
Comment (gsheni) [inline docs/source/release_notes.rst:10]:
```suggestion
        * Updated the conda install commands to specify the channel (:pr:`1917`)
```

**Review by gsheni — APPROVED** [review]

**Comment by gsheni** [comment]
@kushal-gopal Thanks! Congrats on your first contribution to Featuretools!

**Pull request merged**